react-components/table-intro/step9.md

```diff
@@ -1,57 +1,49 @@
-In this step, we'll find and treat a bug that you may or may not have already noticed in the current implementation, and then properly mark our callbacks that use state properties as react "callback hooks" so that they work as expected within the react data change detection ecosystem that is <a href="https://reactjs.org/docs/hooks-intro.html" target="_blank">hooks</a>.
+In this step, we'll add in the functionality of paginating the table's dataset. We'll add a new state property to track the dynamic rows and add a couple more handler functions to facilitate keeping the table and pagination in sync.
 
-## Task
+1) <strong>Create a new state property to keep track of our dynamic rows.</strong>
 
-1) Set the per page dropdown to "2 per page" and navigate to the very last page using the pagination controls, now select "3 per page" from the dropdown and notice our rows are gone!
+Add the following code just below the other state properties you added previously. Set the initial value to the `defaultRows` constant imported from data.js.
 
-2) Locate the `handlePerPageSelect` handler function and replace the call to `updateList(currentPage);` with the following logic.
+<pre class="file">
+const [rows, setRows] = React.useState(defaultRows.slice(0, defaultPerPage));
+</pre>
 
-<pre class="file" data-target="clipboard">
-// when a user reaches end of pagination, and sets numPerPage to a value
-// that requires more data than we have available, reset pagination to page 1
-if (value > numPerPage && currentPage === defaultRows.length / numPerPage) {
-  updateList(1);
-} else {
-  updateList(currentPage);
-}
+2) <strong>Update the `<Table />`'s `rows` prop to use the new `rows` state property.</strong>
+
+Locate the code that sets the rows for the table `rows={defaultRows}`, and update it to reference the new rows state property you just created.
+
+<pre class="file">
+rows={rows}
 </pre>
 
-<strong>Disclaimer: </strong> We've handled the case of pagination overflow here in a basic way to keep this tutorial short and simple. In a production application it might make sense to instead add logic that reverts a user to the last and closest page index that will allow for displaying the number of items per page being selected.
+<strong>Note: </strong> the itemCount prop for the Pagination component will still reference the defaultRows constant as it needs to be aware of the total number of rows in the entire dataset, not the number of rows that's being shown for a given page within the dataset.
```

Comment:
wrap itemCount and defaultRows in back-ticks
Pagination component should be lower case